common/app/http/ResultWithPreload.scala

```diff
@@ -0,0 +1,24 @@
+package http
+
+import model.ApplicationContext
+import play.api.mvc.Result
+
+trait ResultWithPreload {
+  final implicit class RichResult(self: Result) {
+
+    val linkHeaderKey = "Link"
+
+    def withPreload(files: Seq[String])(implicit context: ApplicationContext): Result = {
+      if (files.nonEmpty) {
+        val (jsFiles, cssFiles) = files.partition(_.endsWith(".js"))
+        val preloadFilesCss = cssFiles.map(cssFile =>
+          s"<${conf.Static(common.Assets.css.projectCss(Some(cssFile.stripSuffix(".css"))))}>; rel=preload; as=style; nopush"
+        )
+        val preloadFilesJs = jsFiles.map(file => s"<${conf.Static(file)}>; rel=preload; as=script; nopush")
+        val preloadFiles = (preloadFilesCss ++ preloadFilesJs).mkString(",")
```

Comment:
as mentioned in other comment this code could become:
    
    val preloadFiles = files.map({
        case CssPreloadAsset(...) => ...
        case JsPreloadAsset(...) => ...
    }).mkString(",")